packages/tupaia-web/src/features/View/View.tsx

```diff
@@ -0,0 +1,83 @@
+/**
+ * Tupaia
+ * Copyright (c) 2017 - 2023 Beyond Essential Systems Pty Ltd
+ */
+import React from 'react';
+import { ViewConfig } from '@tupaia/types';
+import { ViewReport } from '../../types';
+import { SingleDownloadLink } from './SingleDownloadLink';
+import { SingleDate } from './SingleDate';
+import { SingleValue } from './SingleValue';
+import { MultiValue } from './MultiValue';
+import { formatDataValueByType } from '@tupaia/utils';
+import { MultiValueRow } from './MultiValueRow';
+
+interface ViewProps {
+  report: ViewReport;
+  config: ViewConfig;
+  isEnlarged?: boolean;
+}
+
+const VIEWS = {
+  singleValue: SingleValue,
+  singleDate: SingleDate,
+  singleDownloadLink: SingleDownloadLink,
+  multiValue: MultiValue,
+  multiValueRow: MultiValueRow,
+};
+
+const formatData = (data: ViewReport['data'], config: ViewConfig) => {
+  const { valueType, value_metadata: valueMetadata } = config;
+  return data?.map(datum => {
+    const { value } = datum;
+    const metadata = {
+      ...(valueMetadata || config[`${datum.name}_metadata`] || {}),
+      ...datum,
+    };
+    return {
+      ...datum,
+      value: formatDataValueByType(
+        {
+          value,
+          metadata,
+        },
+        valueType,
+      ),
+    };
+  });
+};
+
+export const View = ({ report, config, isEnlarged }: ViewProps) => {
```

Comment:
I don't love the name view as it is soo generic it could mean anything but I think it's too baked into the system at this point to rename so I'll have to live with it.

Reply:
I agree - I think in this case it makes sense though, because the `type` of report is `view`